nifi-nar-bundles/nifi-workday-bundle/nifi-workday-processors/src/main/java/org/apache/nifi/processors/workday/GetWorkdayReport.java

```diff
@@ -0,0 +1,432 @@
+        .addValidator(NON_BLANK_VALIDATOR)
+        .build();
+
+    protected static final PropertyDescriptor HASHING_ALGORITHM = new PropertyDescriptor.Builder()
+        .name("Hashing algorithm")
+        .displayName("Hashing algorithm")
+        .description("Determines what hashing algorithm should be used to perform the hashing function.")
+        .required(true)
+        .allowableValues(HashService.buildHashAlgorithmAllowableValues())
+        .defaultValue(HashAlgorithm.SHA256.getName())
+        .addValidator(NON_BLANK_VALIDATOR)
+        .expressionLanguageSupported(NONE)
+        .dependsOn(FIELDS_TO_HASH)
+        .build();
+
+    protected static final PropertyDescriptor RECORD_READER_FACTORY = new PropertyDescriptor.Builder()
+        .name("record-reader")
+        .displayName("Record Reader")
+        .description("Specifies the Controller Service to use for parsing incoming data and determining the data's schema.")
+        .identifiesControllerService(RecordReaderFactory.class)
+        .required(false)
+        .build();
+
+    protected static final PropertyDescriptor RECORD_WRITER_FACTORY = new PropertyDescriptor.Builder()
+        .name("record-writer")
+        .displayName("Record Writer")
+        .description("The Record Writer to use for serializing Records to an output FlowFile.")
+        .identifiesControllerService(RecordSetWriterFactory.class)
+        .dependsOn(RECORD_READER_FACTORY)
+        .required(true)
+        .build();
+
+    protected static final Relationship ORIGINAL = new Relationship.Builder()
+        .name("Original")
+        .description("Request FlowFiles transferred when receiving HTTP responses with a status code between 200 and 299.")
+        .build();
+
+    protected static final Relationship FAILURE = new Relationship.Builder()
+        .name("Failure")
+        .description("Request FlowFiles transferred when receiving socket communication errors.")
+        .build();
+
+    protected static final Relationship RESPONSE = new Relationship.Builder()
+        .name("Response")
+        .description("Response FlowFiles transferred when receiving HTTP responses with a status code between 200 and 299.")
+        .build();
+
+    protected static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(ORIGINAL, RESPONSE, FAILURE)));
+    protected static final List<PropertyDescriptor> PROPERTIES = Collections.unmodifiableList(Arrays.asList(REPORT_URL, WORKDAY_USERNAME, WORKDAY_PASSWORD, WEB_CLIENT_SERVICE,
+        FIELDS_TO_HASH, HASHING_ALGORITHM, RECORD_READER_FACTORY, RECORD_WRITER_FACTORY));
+
+    private final AtomicReference<WebClientService> webClientAtomicReference = new AtomicReference<>();
+    private final AtomicReference<RecordReaderFactory> recordReaderFactoryAtomicReference = new AtomicReference<>();
+    private final AtomicReference<RecordSetWriterFactory> recordSetWriterFactoryAtomicReference = new AtomicReference<>();
```

Comment:
Recommend shortening these names to remove `Atomic`:
```suggestion
    private final AtomicReference<WebClientService> webClientReference = new AtomicReference<>();
    private final AtomicReference<RecordReaderFactory> recordReaderFactoryReference = new AtomicReference<>();
    private final AtomicReference<RecordSetWriterFactory> recordSetWriterFactoryReference = new AtomicReference<>();
```

Reply:
done